lib/openapi_client_generator.rb

```diff
@@ -213,12 +250,9 @@ def method_name
       method_name = case verb
         when "GET"
           namespace
-        when "POST", "PATCH", "DELETE"
-          definition.operation_id.split("/").last.gsub("-", "_")
-        when "PUT"
-          # expecting this to not generalize well, but works for now
+        when "POST", "PATCH", "DELETE", "PUT"
           segments = definition.operation_id.split("/").last.split("-")
-          ([segments.first] + segments[-2..-1]).join("_")
+          segments.size > 4 ? ([segments.first] + segments[-2..-1]).join("_") : "#{action}_#{namespace}"
```

Comment:
Seems like this is a magic number? Should we constantize this?

Reply:
Yeah 😓 It's mostly because I'm guarding against one endpoint that I feel is unnecessarily long: `update_information_about_page_site`. I'll look at it again and see if I can get it to work normally, and if not, I can constantize it 👍